src/MusicStore.Spa/Helpers/AngularExtensions.cs

```diff
@@ -51,20 +51,20 @@ public static class AngularExtensions
             ngAttributes["ng-model"] = valueFieldName;
 
             // Set input type
-            if (string.Equals(metadata.DataTypeName, Enum.GetName(typeof(DataType), DataType.EmailAddress), StringComparison.OrdinalIgnoreCase))
+            if (string.Equals(modelExplorer.Metadata.DataTypeName, Enum.GetName(typeof(DataType), DataType.EmailAddress), StringComparison.OrdinalIgnoreCase))
```

Comment:
nit: use `nameof(DataType.EmailAddress)` instead of `Enum.GetName(typeof(DataType), DataType.EmailAddress)`.  that is, unless Roslyn disallows `nameof` with `enum` values at the moment.